plugin-server/tests/worker/capabilities.test.ts

```diff
@@ -109,6 +109,50 @@ describe('capabilities', () => {
             })
         })
 
+        describe('ingestionOverflow', () => {
```

Comment:
Maybe we should design the test to try all plugin server modes and include a list of exceptions of modes that shouldn't have `processEvent` enabled?

That way, when adding a new mode we will be able to either:
1. Enable `processEvent`.
2. Explicitly add it to the list of exceptions.

Otherwise the test will fail alerting us to potentially missing the capability.

Reply:
Tests look good though, just mentioning this as an alternative to a comment which people may overlook.

Reply:
Agreed: I started with that approach, but I have enough spaghetti in my hands with the PG routing PR that I went to a quick fix here.